packages/app-lib/src/api/profile/mod.rs

```diff
@@ -775,13 +775,30 @@ pub async fn run_credentials(
         ))
     })?;
 
+    let java_args = profile
+    .java
+    .as_ref()
+    .and_then(|it| it.extra_arguments.as_ref())
+    .unwrap_or(&settings.custom_java_args);
+
     let pre_launch_hooks =
         &profile.hooks.as_ref().unwrap_or(&settings.hooks).pre_launch;
     if let Some(hook) = pre_launch_hooks {
-        // TODO: hook parameters
-        let mut cmd = hook.split(' ');
+        // TODO: Improve Hook parameters
+        
+        let full_path = path.get_full_path().await?;
+        let jre_key = get_optimal_jre_key(&path).await.unwrap();
+        let cmd_with_hook_parameters = hook
+            .replace("$INST_NAME", profile.metadata.name.as_str())
+            .replace("$INST_ID", profile.metadata.name.as_str())
+            .replace("$INST_DIR", full_path.to_str().unwrap())
+            .replace("$INST_MC_DIR", full_path.to_str().unwrap())
```

Comment:
There is no difference between `$INST_DIR` and `$INST_MC_DIR`...

Reply:
The reason it's different because on MultiMC-based launchers, usually:

`$INST_DIR` refers to the root instance directory path
`$INST_MC_DIR` refers to the `.minecraft` directory which is usually under the `$INST_DIR`.

For other launchers like ATlauncher, they are the same, if you use `$INST_DIR` or `$INST_MC_DIR`, they do it for compatibility reasons so installing scripts on the instance to launch before or after launching the game would be easier for the end users as most launchers will have something similar.

I would suggest to keep it for two main reasons:

1. Other than compatibility, it makes it easier to import instances into Modrinth Profile from launchers like Prism Launcher, MultiMC, and ATLauncher.
2. It's more future-proof if the project changes or moves how files are stored, for example, if at a later major version, the desktop app stores the data related to the launcher in a directory and Minecraft-related files inside another directory called `.minecraft` or something similar, in that case, the users don't need to update anything from their side, eliminating the need of breaking changes.

If you prefer to remove one of them, then we might want to consider renaming them all as they are no longer compatible with MultiMC-based launchers and similar launchers. 

Maybe something like `$PROFILE_DIR` to avoid confusion and make them specific to **Modrinth Desktop App** and ensure consistency.

Reply:
Okay, so I think we can keep both, but please add a short comment with note describing this situation.

Reply:
> Okay, so I think we can keep both, but please add a short comment with note describing this situation.

Sure, I would also suggest documenting this feature somewhere in Modrinth docs and providing a link in the app.

Reply:
Added a comment, feel free to update it.